### PR TITLE
refactor(mentions): rewrite with hook

### PR DESCRIPTION
**Pull request by hengkx**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  refactor Mentions with hook         |
| 🇨🇳 Chinese | Mentions 用 react hook 重构         |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️ 

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/mentions/index.tsx [hunk 1]
```diff
@@ -38,75 +38,47 @@ interface MentionsEntity {
...
     }
 
     return renderEmpty('Select');
-  }
+  };
 
-  getOptions = () => {
-    const { children, loading } = this.props;
+  const getOptions = () => {
+    const { children } = restProps;
```
Comment (afc163) [inline components/mentions/index.tsx:81]: 这里面的 props 都提到外面去。

**Comment by afc163** [comment]
为啥包体积增加了。

**Comment by hengkx** [comment]
image
看着不像 增加了

**Comment by hengkx** [comment]
image
代码减少了 还增加，，，，

File: components/mentions/index.tsx [hunk 2]
```diff
@@ -52,6 +52,7 @@ const InternalMentions: React.ForwardRefRenderFunction<unknown, MentionProps> =
   const innerRef = React.useRef<HTMLElement>();
   const mergedRef = composeRef(ref, innerRef);
   const { getPrefixCls, renderEmpty, direction } = React.useContext(ConfigContext);
+  const { filterOption, children, notFoundContent } = restProps;
```
Comment (afc163) [inline components/mentions/index.tsx:55]: 直接从 props 里引用出来。

Comment (hengkx) [reply components/mentions/index.tsx:55]: 下面有用到`restProps` ,重新在赋值？

Comment (afc163) [reply components/mentions/index.tsx:55]: 这三个下面都用不到。

**Pull request merged**